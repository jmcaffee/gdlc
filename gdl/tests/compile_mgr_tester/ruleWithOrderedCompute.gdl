/* **************************************************************************
 *  File:
 *
 *  Description:
 *    This is to test a missing rule definition
 *
 ************************************************************************* */


dpm numeric var1;
dpm numeric var2;

rule TestRule()             // Define the rule.
  if(
      ([1]var1 > 100000) &&
      ([2]var2 == 1)
  )
  then
    var1 = var2;
  end
end

ruleset TestRuleset(true)
  rule TestRule();         // Rule was defined above.
end

