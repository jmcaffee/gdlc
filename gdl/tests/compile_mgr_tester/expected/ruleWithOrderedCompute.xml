<Rule Id="1" Name="TestRule"><IfMessages/><ElseMessages/><Condition><Expression><Compute><Operator>AND</Operator><LeftOperand><Brace><Compute Order="1"><Operator>&gt;</Operator><LeftOperand><DPM Name="var1"/></LeftOperand><RightOperand><Constant>100000</Constant></RightOperand></Compute></Brace></LeftOperand><RightOperand><Brace><Compute Order="2"><Operator>==</Operator><LeftOperand><DPM Name="var2"/></LeftOperand><RightOperand><Constant>1</Constant></RightOperand></Compute></Brace></RightOperand></Compute></Expression></Condition><IfActions><Assign><AssignTo><DPM Name="var1"/></AssignTo><AssignValue><Expression><DPM Name="var2"/></Expression></AssignValue></Assign></IfActions></Rule>